### PR TITLE
Sanitize defaults

### PR DESCRIPTION
**Pull request by sol**
Addresses #244.

### CODE REVIEW

File: src/Hpack/Syntax.hs [hunk 1]
```diff
@@ -1,40 +1,110 @@
...
+        | isValidRef ref = return (Ref ref)
+        | otherwise = fail ("invalid reference " ++ show ref)
+
+data Path = Path {unPath :: [FilePath]}
+
+instance FromJSON Path where
+  parseJSON v = parseJSON v >>= parsePath
+    where
+      parsePath path
+        | '\\' `elem` path = fail ("rejecting '\\' in " ++ show path ++ ", please use '/' to separate path components")
```
Comment (quasicomputational) [inline src/Hpack/Syntax.hs]: Backslash is a perfectly cromulent path character in POSIX, though I guess there's not too much harm in subsetting for Windows compatibility.

File: src/Hpack/Syntax.hs [hunk 2]
```diff
@@ -1,40 +1,110 @@
...
+        | "/" `elem` p = fail ("rejecting absolute path " ++ show path)
+        | ".." `elem` p = fail ("rejecting \"..\" in " ++ show path)
+        | otherwise = return (Path p)
```
Comment (quasicomputational) [inline src/Hpack/Syntax.hs]: Are we sure there are no weird Windows corner cases here? I'd also want to double check that `~` won't get expanded to `$HOME`.

File: src/Hpack/Syntax.hs (hunk 1 above)
Comment (quasicomputational) [reply src/Hpack/Syntax.hs]: Forgot to say that, if we do subset POSIX paths for compatibility, that should be documented.

File: src/Hpack/Syntax.hs (hunk 2 above)
Comment (sol) [reply src/Hpack/Syntax.hs]: > Are we sure there are no weird Windows corner cases here?

Apparently we need to exclude `:`.

> I'd also want to double check that ~ won't get expanded to $HOME.

It won't, `~` is expanded by the shell.

**Pull request merged**